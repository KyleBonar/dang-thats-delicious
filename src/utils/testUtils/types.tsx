export interface MockLink {
  text: string;
  to: string;
  target: "_blank" | "_self";
}
