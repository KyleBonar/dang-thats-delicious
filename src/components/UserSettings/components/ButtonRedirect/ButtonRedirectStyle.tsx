import styled from "../../../../theme/styled-components";
import { Button } from "../../../Button/Button";

export const StyledButton = styled(Button)`
  display: inline-block;

  > button {
    color: #333;
  }
`;
StyledButton.displayName = "Button";

export const StyledDiv = styled.div`
  margin-bottom: 32px;
`;
StyledDiv.displayName = "div";
