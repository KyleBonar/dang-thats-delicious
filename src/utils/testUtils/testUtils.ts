import * as React from "react";
import casual from "casual";
import { ReactWrapper } from "enzyme";
import configureStore from "redux-mock-store";

import { MockCard } from "./types";
import { AppState } from "../../redux/configureStore";
import { ISauce, ISaucesState } from "../../redux/sauces/types";
import { FlashMessageProps } from "../../components/FlashMessage/FlashMessage";
import Flatn from "../Flatn/Flatn";
import { IUserState } from "../../redux/users/types";

export const ITERATION_SIZE = 32;
const REACT_REGEX = /react(\d+)?./i;

// seed it so we get consistent results
casual.seed(777);

const relatedSauce = () => ({
  name: casual.name,
  slug: casual.uuid
});

export const fakeSauce = (): ISauce => ({
  _id: casual.random_element([undefined, casual.integer(1, 500)]),
  _addedToStore: casual.random_element([undefined, casual.unix_time]), // Unix time (in seconds) added to redux store
  _full: casual.random_element([undefined, casual.boolean]), // Whether we have full sauce or partial
  _related: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(relatedSauce)
  ]), // List of related sauces
  isAdminApproved: casual.random_element([undefined, casual.boolean]),
  name: casual.name,
  ingredients: casual.string,
  author: casual.name,
  created: casual.unix_time, // Unix time (in seconds)
  types: casual.array_of_words(),
  maker: casual.name,
  description: casual.string,
  photo: casual.random_element([undefined, casual.url]),
  shu: casual.random_element([
    undefined,
    casual.integer(1, 500),
    casual.integer(1, 500).toString()
  ]),
  reviews: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(x => casual.word)
  ]),
  city: casual.random_element([undefined, casual.city]),
  state: casual.random_element([undefined, casual.state]),
  country: casual.random_element([undefined, casual.country]),
  slug: casual.random_element([undefined, casual.uuid])
});

export const fakeSaucesState = (): ISaucesState => {
  const sauces: ISauce[] = new Array(casual.integer(1, 15))
    .fill(null)
    .map(fakeSauce);
  const { allSlugs, bySlug } = Flatn.saucesArr({ sauces });

  return {
    allSlugs,
    bySlug,
    total: allSlugs.length,
    query: undefined,
    saucesWithNewestReviews: undefined,
    newest: undefined,
    featured: undefined,
    types: casual.array_of_words(),
    orders: ["Newest", "Name", "Times Reviewed", "Avg Rating"]
  };
};

const fakeUsersState = (): IUserState => ({
  self: casual.random_element([
    undefined,
    {
      token: casual.random_element([undefined, casual.uuid]),
      displayName: casual.random_element([undefined, casual.name]),
      avatarURL: casual.random_element([undefined, casual.url]),
      isAdmin: casual.random_element([undefined, casual.boolean])
    }
  ]),
  byDisplayName: undefined,
  allDisplayNames: undefined
});

export const fakeStore = () => {
  const storeConfig = configureStore([]);

  const store: AppState = {
    sauces: fakeSaucesState(),
    users: fakeUsersState(),
    reviews: {
      byReviewID: undefined,
      allReviewIDs: undefined
    }
  };

  return storeConfig(store);
};

export const fakeJSXElement = (): JSX.Element => {
  return casual.random_element([
    React.createElement("div", { key: casual.uuid }, casual.string),
    React.createElement("span", { key: casual.uuid }, casual.string)
  ]);
};

const fakeCard = (): MockCard => ({
  title: casual.string,
  description: casual.string,
  to: casual.url,
  showLink: casual.boolean,
  imageLink: casual.url,
  className: casual.string,
  anchorText: casual.string
});

const wait = (amount = 0) =>
  new Promise(resolve => setTimeout(resolve, amount));

function simulateInputChange(
  wrapper: ReactWrapper,
  name: string,
  value: string
) {
  wrapper
    // .find(`input[name='${name}']`)
    .simulate("change", { target: { name, value } });
}

const isClassComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    component.prototype &&
    !!component.prototype.isReactComponent
  );
};

// Ensure compatability with transformed code
const isFunctionComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    String(component).includes("return") &&
    !!String(component).match(REACT_REGEX) &&
    String(component).includes(".createElement")
  );
};

const isElement = (typeElement): boolean => {
  return React.isValidElement(typeElement);
};

const isComponent = (component): boolean => {
  return isClassComponent(component) || isFunctionComponent(component);
};

const isDOMTypeElement = (typeElement): boolean => {
  return isElement(typeElement) && typeof typeElement.type === "string";
};

const generateValidPassword = (MIN_PASSWORD_LENGTH: number = 8) => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(
    MIN_PASSWORD_LENGTH,
    3 * MIN_PASSWORD_LENGTH
  );

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

const generateInValidPassword = (MIN_PASSWORD_LENGTH: number = 8) => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(0, MIN_PASSWORD_LENGTH);

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

export {
  casual,
  simulateInputChange,
  isDOMTypeElement,
  isComponent,
  wait,
  generateValidPassword
};
