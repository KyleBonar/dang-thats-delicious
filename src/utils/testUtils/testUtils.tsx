import * as React from "react";
import * as enzyme from "enzyme";
import casual from "casual";
import { ReactWrapper } from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import { AppState } from "../../redux/configureStore";
import { ISauce, ISaucesState } from "../../redux/sauces/types";
import Flatn from "../Flatn/Flatn";
import { IUserState } from "../../redux/users/types";
import { IErrReturn } from "../Err/Err";
import { act } from "react-dom/test-utils";
import { Provider } from "react-redux";

export const ITERATION_SIZE = 32;
const REACT_REGEX = /react(\d+)?./i;

// seed it so we get consistent results
casual.seed(777);

const relatedSauce = () => ({
  name: casual.name,
  slug: casual.uuid
});

export const fakeSauce = (): ISauce => ({
  _id: casual.random_element([undefined, casual.integer(1, 500)]),
  _addedToStore: casual.random_element([undefined, casual.unix_time]), // Unix time (in seconds) added to redux store
  _full: casual.random_element([undefined, casual.boolean]), // Whether we have full sauce or partial
  _related: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(relatedSauce)
  ]), // List of related sauces
  isAdminApproved: casual.random_element([undefined, casual.boolean]),
  name: casual.name,
  ingredients: casual.random_element([undefined, casual.string]),
  author: casual.name,
  created: casual.unix_time, // Unix time (in seconds)
  types: casual.random_element([undefined, casual.array_of_words()]),
  maker: casual.name,
  description: casual.string,
  photo: casual.random_element([undefined, casual.url]),
  shu: casual.random_element([
    undefined,
    casual.integer(1, 500),
    casual.integer(1, 500).toString()
  ]),
  reviews: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(x => casual.word)
  ]),
  city: casual.random_element([undefined, casual.city]),
  state: casual.random_element([undefined, casual.state]),
  country: casual.random_element([undefined, casual.country]),
  slug: casual.random_element([undefined, casual.uuid])
});

export const generateFakeSauces = (): ISauce[] =>
  new Array(casual.integer(1, 15)).fill(null).map(fakeSauce);

export const fakeSaucesState = (): ISaucesState => {
  const sauces: ISauce[] = generateFakeSauces();
  const { allSlugs, bySlug } = Flatn.saucesArr({ sauces });

  const featured: undefined | string[] = casual.random_element([
    undefined,
    allSlugs.length > 0
      ? allSlugs.filter(slug => {
          if (casual.random < 0.4) return slug;
          return;
        })
      : undefined
  ]);

  return {
    allSlugs,
    bySlug,
    total: allSlugs.length,
    query: undefined,
    saucesWithNewestReviews: undefined,
    newest: undefined,
    featured,
    types: casual.array_of_words(),
    orders: ["Newest", "Name", "Times Reviewed", "Avg Rating"]
  };
};

const fakeUsersState = (): IUserState => ({
  self: casual.random_element([
    undefined,
    {
      token: casual.random_element([undefined, casual.uuid]),
      displayName: casual.random_element([undefined, casual.name]),
      avatarURL: casual.random_element([undefined, casual.url]),
      isAdmin: casual.random_element([undefined, casual.boolean])
    }
  ]),
  byDisplayName: undefined,
  allDisplayNames: undefined
});

export const fakeStore = () => {
  const middlewares = [thunk];
  const storeConfig = configureStore(middlewares);

  const store: AppState = {
    sauces: fakeSaucesState(),
    users: fakeUsersState(),
    reviews: {
      byReviewID: undefined,
      allReviewIDs: undefined
    }
  };

  return storeConfig(store);
};

export const fakeJSXElement = (): JSX.Element => {
  return casual.random_element([
    React.createElement("div", { key: casual.uuid }, casual.string),
    React.createElement("span", { key: casual.uuid }, casual.string)
  ]);
};

const wait = (amount = 0) =>
  new Promise(resolve => setTimeout(resolve, amount));

function simulateInputChange(
  wrapper: ReactWrapper,
  name: string,
  value: string
) {
  wrapper
    // .find(`input[name='${name}']`)
    .simulate("change", { target: { name, value } });
}

const isClassComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    component.prototype &&
    !!component.prototype.isReactComponent
  );
};

// Ensure compatability with transformed code
const isFunctionComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    String(component).includes("return") &&
    !!String(component).match(REACT_REGEX) &&
    String(component).includes(".createElement")
  );
};

const isElement = (typeElement): boolean => {
  return React.isValidElement(typeElement);
};

const isComponent = (component): boolean => {
  return isClassComponent(component) || isFunctionComponent(component);
};

const isDOMTypeElement = (typeElement): boolean => {
  return isElement(typeElement) && typeof typeElement.type === "string";
};

const generateValidPassword = (MIN_PASSWORD_LENGTH: number = 8): string => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(
    MIN_PASSWORD_LENGTH,
    3 * MIN_PASSWORD_LENGTH
  );

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

const generateInValidPassword = (MIN_PASSWORD_LENGTH: number = 8): string => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(1, MIN_PASSWORD_LENGTH - 1);

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

// Function designed to resemble how a network error would look like.
// This will help the catch(err) only have to handle a single format
const generateErr = (): IErrReturn => {
  return {
    response: {
      data: {
        status: 400,
        msg: casual.string,
        isGood: false
      }
    }
  };
};

// lets us mount up a simple custom hook
const mountReactHook = hook => {
  const Component = ({ children }) => children(hook());
  const componentHook = {};
  let componentMount;

  act(() => {
    componentMount = enzyme.shallow(
      <Component>
        {hookValues => {
          Object.assign(componentHook, hookValues);
          return null;
        }}
      </Component>
    );
  });
  return { componentMount, componentHook };
};

export interface ImountReactHookWithReduxStore {
  componentMount: any;
  componentHook: any;
}
const mountReactHookWithReduxStore = (
  hook,
  store
): ImountReactHookWithReduxStore => {
  const Component = ({ children }) => children(hook());
  const componentHook = {};
  let componentMount;

  act(() => {
    componentMount = enzyme.mount(
      <Provider store={store}>
        <Component>
          {hookValues => {
            Object.assign(componentHook, hookValues);
            return null;
          }}
        </Component>
      </Provider>
    );
  });
  return { componentMount, componentHook };
};

export {
  casual,
  simulateInputChange,
  isDOMTypeElement,
  isComponent,
  wait,
  generateValidPassword,
  generateInValidPassword,
  generateErr,
  mountReactHook,
  mountReactHookWithReduxStore
};
