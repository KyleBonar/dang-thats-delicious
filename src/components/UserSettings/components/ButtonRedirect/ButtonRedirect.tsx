import * as React from "react";

import { StyledButton, StyledDiv } from "./ButtonRedirectStyle";
import ArrowRight from "../../../../images/icons/ArrowRight";
import { Link } from "../../../Link/Link";

interface IButtonRedirectProps {
  name: string;
  href: string;
}

const ButtonRedirect: React.FC<IButtonRedirectProps> = props => {
  return (
    <StyledDiv>
      <h4>{props.name}</h4>
      <Link href={`${props.href}`}>
        <StyledButton type="button">
          {props.name} <ArrowRight />
        </StyledButton>
      </Link>
    </StyledDiv>
  );
};

export default ButtonRedirect;
