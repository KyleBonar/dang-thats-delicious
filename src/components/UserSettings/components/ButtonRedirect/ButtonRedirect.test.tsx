import * as React from "react";
import * as enzyme from "enzyme";

import ButtonRedirect, { IButtonRedirectProps } from "./ButtonRedirect";
import { casual, ITERATION_SIZE } from "../../../../utils/testUtils/testUtils";

const fakeButtonRedirect = (): IButtonRedirectProps => ({
  name: casual.string,
  href: casual.url
});

const mockButtonRedirects = new Array(ITERATION_SIZE)
  .fill(null)
  .map(fakeButtonRedirect);

describe("<ButtonRedirect />", () => {
  it("renders", () => {
    mockButtonRedirects.forEach(mockButtonRedirect => {
      const wrapper = enzyme.shallow(
        <ButtonRedirect {...mockButtonRedirect} />
      );

      expect(wrapper).toBeTruthy();
    });
  });

  it("matches snapshot", () => {
    mockButtonRedirects.forEach(mockButtonRedirect => {
      const wrapper = enzyme.shallow(
        <ButtonRedirect {...mockButtonRedirect} />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
