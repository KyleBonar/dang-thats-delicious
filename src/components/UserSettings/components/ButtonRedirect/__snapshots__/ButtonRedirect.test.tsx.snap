// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonRedirect /> matches snapshot 1`] = `
<div>
  <h4>
    sequi facilis corporis sed veritatis quidem praesentium
  </h4>
  <Link
    href="http://www.Luettgen.tv/"
  >
    <Button
      type="button"
    >
      sequi facilis corporis sed veritatis quidem praesentium
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 2`] = `
<div>
  <h4>
    laborum est possimus assumenda aut enim iure
  </h4>
  <Link
    href="http://Augustus.biz/"
  >
    <Button
      type="button"
    >
      laborum est possimus assumenda aut enim iure
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 3`] = `
<div>
  <h4>
    vitae occaecati officia iure nisi ut maiores
  </h4>
  <Link
    href="http://Howell.io/"
  >
    <Button
      type="button"
    >
      vitae occaecati officia iure nisi ut maiores
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 4`] = `
<div>
  <h4>
    non omnis consequatur enim quia autem et
  </h4>
  <Link
    href="http://www.Dibbert.net/"
  >
    <Button
      type="button"
    >
      non omnis consequatur enim quia autem et
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 5`] = `
<div>
  <h4>
    reprehenderit omnis adipisci eius laborum eius aut
  </h4>
  <Link
    href="http://www.Beahan.net/"
  >
    <Button
      type="button"
    >
      reprehenderit omnis adipisci eius laborum eius aut
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 6`] = `
<div>
  <h4>
    id sit delectus vel ipsa tempora aut
  </h4>
  <Link
    href="http://Joshua.biz/"
  >
    <Button
      type="button"
    >
      id sit delectus vel ipsa tempora aut
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 7`] = `
<div>
  <h4>
    animi vel provident dicta perferendis dolorum cupiditate
  </h4>
  <Link
    href="http://Turcotte.org/"
  >
    <Button
      type="button"
    >
      animi vel provident dicta perferendis dolorum cupiditate
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 8`] = `
<div>
  <h4>
    sint quia sed exercitationem omnis laborum exercitationem
  </h4>
  <Link
    href="http://Klocko.info/"
  >
    <Button
      type="button"
    >
      sint quia sed exercitationem omnis laborum exercitationem
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 9`] = `
<div>
  <h4>
    assumenda quasi adipisci mollitia iste itaque omnis
  </h4>
  <Link
    href="http://Kevin.biz/"
  >
    <Button
      type="button"
    >
      assumenda quasi adipisci mollitia iste itaque omnis
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 10`] = `
<div>
  <h4>
    aliquam sed sapiente soluta corporis commodi eaque
  </h4>
  <Link
    href="http://www.Kiehn.ca/"
  >
    <Button
      type="button"
    >
      aliquam sed sapiente soluta corporis commodi eaque
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 11`] = `
<div>
  <h4>
    doloremque maiores esse atque et non explicabo
  </h4>
  <Link
    href="http://www.Eva.info/"
  >
    <Button
      type="button"
    >
      doloremque maiores esse atque et non explicabo
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 12`] = `
<div>
  <h4>
    itaque voluptas ea ducimus autem sit aspernatur
  </h4>
  <Link
    href="http://www.Mathilde.org/"
  >
    <Button
      type="button"
    >
      itaque voluptas ea ducimus autem sit aspernatur
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 13`] = `
<div>
  <h4>
    fugiat dolorem nihil quam nulla est unde
  </h4>
  <Link
    href="http://www.Madge.org/"
  >
    <Button
      type="button"
    >
      fugiat dolorem nihil quam nulla est unde
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 14`] = `
<div>
  <h4>
    doloremque explicabo eaque necessitatibus sit suscipit omnis
  </h4>
  <Link
    href="http://Greenholt.com/"
  >
    <Button
      type="button"
    >
      doloremque explicabo eaque necessitatibus sit suscipit omnis
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 15`] = `
<div>
  <h4>
    corporis qui aut nobis quo consequatur harum
  </h4>
  <Link
    href="http://Jewel.co.uk/"
  >
    <Button
      type="button"
    >
      corporis qui aut nobis quo consequatur harum
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 16`] = `
<div>
  <h4>
    voluptate in sunt occaecati nulla aut sunt
  </h4>
  <Link
    href="http://Steuber.info/"
  >
    <Button
      type="button"
    >
      voluptate in sunt occaecati nulla aut sunt
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 17`] = `
<div>
  <h4>
    dolores fugiat quis et voluptas culpa et
  </h4>
  <Link
    href="http://Tromp.org/"
  >
    <Button
      type="button"
    >
      dolores fugiat quis et voluptas culpa et
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 18`] = `
<div>
  <h4>
    quis quae iusto sapiente nostrum aut corrupti
  </h4>
  <Link
    href="http://www.Marilie.com/"
  >
    <Button
      type="button"
    >
      quis quae iusto sapiente nostrum aut corrupti
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 19`] = `
<div>
  <h4>
    laudantium dolor sint officiis qui dignissimos dolore
  </h4>
  <Link
    href="http://www.Maximus.io/"
  >
    <Button
      type="button"
    >
      laudantium dolor sint officiis qui dignissimos dolore
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 20`] = `
<div>
  <h4>
    iste est unde rem similique et maxime
  </h4>
  <Link
    href="http://www.Milton.com/"
  >
    <Button
      type="button"
    >
      iste est unde rem similique et maxime
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 21`] = `
<div>
  <h4>
    consequatur sint provident aut ullam distinctio autem
  </h4>
  <Link
    href="http://www.Rosenbaum.biz/"
  >
    <Button
      type="button"
    >
      consequatur sint provident aut ullam distinctio autem
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 22`] = `
<div>
  <h4>
    ut odit animi in ut occaecati at
  </h4>
  <Link
    href="http://www.Ruben.net/"
  >
    <Button
      type="button"
    >
      ut odit animi in ut occaecati at
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 23`] = `
<div>
  <h4>
    occaecati illum laborum quos eum voluptate sit
  </h4>
  <Link
    href="http://McCullough.biz/"
  >
    <Button
      type="button"
    >
      occaecati illum laborum quos eum voluptate sit
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 24`] = `
<div>
  <h4>
    itaque et in fugit in rerum libero
  </h4>
  <Link
    href="http://www.Dayana.name/"
  >
    <Button
      type="button"
    >
      itaque et in fugit in rerum libero
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 25`] = `
<div>
  <h4>
    aut inventore aut eveniet beatae incidunt ut
  </h4>
  <Link
    href="http://www.Kunde.com/"
  >
    <Button
      type="button"
    >
      aut inventore aut eveniet beatae incidunt ut
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 26`] = `
<div>
  <h4>
    aut quidem voluptatibus libero aut autem ut
  </h4>
  <Link
    href="http://www.Rowe.info/"
  >
    <Button
      type="button"
    >
      aut quidem voluptatibus libero aut autem ut
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 27`] = `
<div>
  <h4>
    fugiat maiores autem est esse quos minus
  </h4>
  <Link
    href="http://Bernier.org/"
  >
    <Button
      type="button"
    >
      fugiat maiores autem est esse quos minus
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 28`] = `
<div>
  <h4>
    quo sunt possimus earum animi quis expedita
  </h4>
  <Link
    href="http://Schulist.us/"
  >
    <Button
      type="button"
    >
      quo sunt possimus earum animi quis expedita
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 29`] = `
<div>
  <h4>
    voluptate necessitatibus excepturi et qui sunt nesciunt
  </h4>
  <Link
    href="http://Jean.info/"
  >
    <Button
      type="button"
    >
      voluptate necessitatibus excepturi et qui sunt nesciunt
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 30`] = `
<div>
  <h4>
    rerum nobis nam dolor quas repellendus in
  </h4>
  <Link
    href="http://www.Lazaro.us/"
  >
    <Button
      type="button"
    >
      rerum nobis nam dolor quas repellendus in
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 31`] = `
<div>
  <h4>
    occaecati et consequatur consectetur sunt molestiae cupiditate
  </h4>
  <Link
    href="http://www.Ramiro.co.uk/"
  >
    <Button
      type="button"
    >
      occaecati et consequatur consectetur sunt molestiae cupiditate
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;

exports[`<ButtonRedirect /> matches snapshot 32`] = `
<div>
  <h4>
    molestiae non sint et ipsa consequuntur quidem
  </h4>
  <Link
    href="http://Freeman.name/"
  >
    <Button
      type="button"
    >
      molestiae non sint et ipsa consequuntur quidem
       
      <ArrowRight />
    </Button>
  </Link>
</div>
`;
